File: src/main/java/no/nav/pto_schema/kafka/json/topic/SisteTilordnetVeilederV1.java
package no.nav.pto_schema.kafka.json.topic;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.NoArgsConstructor;
import lombok.Value;

import java.time.ZonedDateTime;

@Value
@NoArgsConstructor(force = true)
@AllArgsConstructor
@Builder
public class SisteTilordnetVeilederV1 {
    String aktorId;
    String veilederId;
    ZonedDateTime tilordnetTidspunkt;
}

File: src/main/kotlin/no/nav/pto_schema/kafka/json/topic/SisteTilordnetVeilederV1.kt
package no.nav.pto_schema.kafka.json.topic

import java.time.ZonedDateTime

data class SisteTilordnetVeilederV1(
    val aktorId: String? = null,
    val veilederId: String? = null,
    val tilordnet: ZonedDateTime? = null,
) {
    companion object {
        @JvmStatic
        fun builder(): Builder = Builder()
    }

    class Builder {
        private var aktorId: String? = null
        private var veilederId: String? = null
        private var tilordnet: ZonedDateTime? = null

        fun aktorId(aktorId: String?): Builder = apply { this.aktorId = aktorId }
        fun veilederId(veilederId: String?): Builder = apply { this.veilederId = veilederId }
        fun tilordnet(tilordnet: ZonedDateTime?): Builder = apply { this.tilordnet = tilordnet }

        fun build(): SisteTilordnetVeilederV1 = SisteTilordnetVeilederV1(aktorId, veilederId, tilordnet)
    }
}

